### PR TITLE
feat: Dockerfile has been added along with the required documentation

### PR DESCRIPTION
**Pull request by wolfcito**
**Description**

This pull request introduces a `Dockerfile` to ensure that **Supersim** can be easily built and run within a containerized environment. By Dockerizing Supersim, we improve the consistency and portability of the development environment, allowing contributors to work on the tool without worrying about dependencies or system conflicts.

The following changes were made:
- Added a `Dockerfile` to define the containerized setup for Supersim.
- Updated the `README.md` with instructions for building and running Supersim using Docker. The instructions now include:
  - `docker build -t supersim .`
  - `docker run --rm -it --network host supersim:latest`
  
This will allow for faster onboarding and easier development across different systems by encapsulating all dependencies within the container.

**Tests**

No new tests have been added for this Docker integration. However, the Dockerfile has been manually tested on different environments (Windows WSL and macOS) to ensure that the build and run processes work as expected. Further automated testing can be added for CI/CD integration in future iterations.

**Additional context**

This change is highly beneficial for developers who work across different ecosystems or environments. It ensures that Supersim can be easily run without needing manual dependency management, thus avoiding potential issues caused by environment discrepancies. The containerized environment guarantees that all necessary dependencies are isolated and controlled.

### CODE REVIEW

**Comment by roninjin10** [comment]
Nice I was gonna open an issue for this!

File: README.md [hunk 1]
```diff
@@ -58,6 +58,26 @@ L2s: Predeploy Contracts Spec ( https://specs.optimism.io/protocol/predeploys.ht
...
+
+### 2. Run Supersim in Docker
+   ```bash
+   docker run --rm -it --network host supersim:latest
+   ```
+
+This setup encapsulates Supersim in a container, isolating it from the host system. This ensures a clean, conflict-free environment and provides an added layer of security by reducing the application's permissions on the host.
+
+---
+
```
Comment (tremarkley) [inline README.md:80]: I think it would be best to have this in its own section. do you mind putting it after the Getting Started section?

Comment (wolfcito) [reply README.md]: Sure thing, no problem. It's done!

**Comment by wolfcito** [comment]
Hello @tremarkley. I'm noticing that the test is still failing, even though I haven't touched any functionality.

If there's any way I can help, please let me know.

**Comment by tremarkley** [comment]
Hey @wolfcito the CI checks for your pr are passing now. Sorry about that, this is our first external pr so had to make some changes to get forks working properly with CI. Going to merge your pr now.

**Pull request merged**